lecture_2/Exercise_2/main.c

```diff
@@ -0,0 +1,21 @@
+#include <stdio.h>
+#include <stdint.h>
+
+void convertRgb888ToRgb656(uint8_t *r,
+                           uint8_t *g,
+                           uint8_t *b)
+{
+    *r = *r >> 2;
+    *g = *g >> 3;
+    *b = *b >> 2;
+}
+
+int main()
+{
+    uint8_t r, g, b;
+    r = g = b =255;
+    convertRgb888ToRgb656(&r, &g, &b);
+    printf("r=%x g=%x b=%x", r, g, b);
+    
+    return 0;
```

Comment:
Ok, in general this program works correctly. 
The problem is that, usually RGB888 packet in one uint32_t variable but not in separate uint8_t variables and RGB565 packed in uint16_t variable.

Sorry, that is my mistake. I did not describe the task clear enough. But any way you have to rework it ;)